allen-cell-animated/simularium-website
# Remove unused dependencies

Problem
=======
Closes #211 

Solution
========
I went through all the dependencies in package.json and removed any that were not imported into other files or otherwise used in this project (e.g. config files).

I used [depcheck](https://www.npmjs.com/package/depcheck) to try to automatically find unused dependencies. It's not perfect, so the list it provided was more of a rough guide. 

`depcheck` on main branch:
```
Unused dependencies
* @aics/simularium-observables-manager
* @types/victory
* momentjs
* rc-motion
* react-player-controls
* url-loader
* victory
Unused devDependencies
* @babel/cli
* @types/chai
* @types/enzyme
* @types/jest
* @types/jsdom
* @types/mocha
* babel-loader
* css-loader
* enzyme
* enzyme-adapter-react-16
* eslint-plugin-react-hooks
* file-loader
* husky
* identity-obj-proxy
* jsdom
* jsdom-global
* less
* less-loader
* lint-staged
* mock-css-modules
* postcss-loader
* prettier
* ts-node
* webpack-cli
Missing dependencies
* moment: ./src/components/ViewerTitle/index.tsx
```

`depcheck` on this branch (anything without a comment is used):
```
Unused dependencies
* url-loader
Unused devDependencies
* @babel/cli   // I don't think this is used but a babel/cli sounds relevant?
* @types/enzyme    // Enzyme is not currently used but we might try it out very soon?
* @types/jest    // We'll probably use this
* @types/jsdom    // We'll probably use this
* babel-loader 
* css-loader  
* enzyme    // Not currently used but might try it out soon
* eslint-plugin-react-hooks    // We're not using this anywhere but we should configure/use this
* file-loader
* husky
* identity-obj-proxy
* jsdom
* jsdom-global
* less
* less-loader
* lint-staged
* postcss-loader
* prettier
* ts-node    // I'm not positive we use this but it sounds too relevant?
* webpack-cli    // We probably need this?
```

## Type of change
Please delete options that are not relevant.

* Maintenance, should not break anything

Steps to Verify:
----------------
1. Pull this branch and `npm i`
2. try `npm start` and make sure site builds and functions as before
3. try `npm run test` and `npm run typeCheck` and make sure nothing's broken
